build/dsc/resource.rb

```diff
@@ -116,13 +116,18 @@ def has_credential?
 
     def ps_module
       unless @ps_module
+
         path_array = @resource_mof_path.split('/')
         revert_array = path_array.reverse
         downcased_array = revert_array.collect{|p| p.downcase}
         index = downcased_array.index('dscresources')
-        raise "module for #{self.name} not found" if index == nil
+        raise "module for #{self.name} not found (Missing DSCResources directory in path #{@resource_mof_path})" if index == nil
 
         module_name = revert_array[index + 1 ] rescue nil
+        if (module_name == "dsc_resources") 
+          warn("The name of the module in directory #{@resource_mof_path} was detected as #{module_name}.  This may indicate that the wrong directory was used to import the resource")
```

Comment:
I think @Iristyle is suggesting that because `if (module_name == 'dsc_resources')` already, do you really need to interpolate that variable in the warning because it would only be equal to `dsc_resources` currently. It's a micro optimization as a review point.